### DISCUSSION TITLE
cutback_bend90 (straight function cannot to used together)

### DISCUSSION THREAD

**Discussion by Rammohan97-boyapati** [gdsfactory/gdsfactory#1060] [Q&A] [answered]
```
import gdsfactory as gf
from gdsfactory.pdk import get_component

d = gf.components.bezier(control_points=[[0.0, 0.0], [6.335, 0.0], [10, 4.665], [10, 10.0]], npoints=201, with_manhattan_facing_angles=True, with_bbox=True, cross_section='Lowrib')
d.plot()
b = gf.components.straight( length = 10,  npoints=2, with_bbox=True, cross_section='Lowrib')
b.plot()
c = gf.components.cutback_bend90(bend90= b, straight = b, straight_length=10, rows=8, columns=3, spacing=20)
c.plot()

```

I am getting this error 

      6 b = gf.components.straight( length = 10,  npoints=2, with_bbox=True, cross_section='Lowrib')
      7 b.plot()
----> 8 c = gf.components.cutback_bend90(bend90= b, straight = b, straight_length=10, rows=8, columns=3, spacing=20)
      9 c.plot()

~\anaconda3\lib\site-packages\gdsfactory\cell.py in _cell(*args, **kwargs)
    159             )
    160 
--> 161         component = func(*args, **kwargs)
    162 
    163         # if the component is already in the cache, but under a different alias,

~\anaconda3\lib\site-packages\pydantic\decorator.cp39-win_amd64.pyd in pydantic.decorator.validate_arguments.validate.wrapper_function()

~\anaconda3\lib\site-packages\pydantic\decorator.cp39-win_amd64.pyd in pydantic.decorator.ValidatedFunction.call()

~\anaconda3\lib\site-packages\pydantic\decorator.cp39-win_amd64.pyd in pydantic.decorator.ValidatedFunction.execute()

~\anaconda3\lib\site-packages\gdsfactory\components\cutback_bend.py in cutback_bend90(bend90, straight_length, rows, columns, spacing, straight, **kwargs)
    104 
    105     bend90 = get_component(bend90, **kwargs)
--> 106     straightx = get_component(straight, length=straight_length, **kwargs)
    107     straight_length = 2 * _get_bend_size(bend90) + spacing + straight_length
    108     straighty = get_component(straight, length=straight_length, **kwargs)

~\anaconda3\lib\site-packages\gdsfactory\pdk.py in get_component(component, **kwargs)
    485 
    486 def get_component(component: ComponentSpec, **kwargs) -> Component:
--> 487     return _ACTIVE_PDK.get_component(component, **kwargs)
    488 
    489 

~\anaconda3\lib\site-packages\gdsfactory\pdk.py in get_component(self, component, **kwargs)
    287     def get_component(self, component: ComponentSpec, **kwargs) -> Component:
    288         """Returns component from a component spec."""
--> 289         return self._get_component(
    290             component=component, cells=self.cells, containers=self.containers, **kwargs
    291         )

~\anaconda3\lib\site-packages\gdsfactory\pdk.py in _get_component(self, component, cells, containers, **kwargs)
    314         if isinstance(component, Component):
    315             if kwargs:
--> 316                 raise ValueError(f"Cannot apply kwargs {kwargs} to {component.name!r}")
    317             return component
    318         elif callable(component):

ValueError: Cannot apply kwargs {'length': 10.0} to 'straight_eef4eda3'

Can you please tell how i can use straight function in cutback_bend90

**Comment by joamatab** [comment, marked as answer]
Hi @ramism 

`cutback_bend90` is expecting a bend pcell function and your are passing a bend Component,

You can checkout this tutorial 
https://gdsfactory.github.io/gdsfactory/notebooks/04_components_hierarchy.html

And this is what I think you are trying to do

```
import gdsfactory as gf

cross_section = "rib"

b = gf.components.bezier(
    control_points=[[0.0, 0.0], [6.335, 0.0], [10, 4.665], [10, 10.0]],
    npoints=201,
    with_manhattan_facing_angles=True,
    with_bbox=True,
    cross_section=cross_section,
) # b is a bend Component

b_function = gf.partial(
    gf.components.bezier,
    control_points=[[0.0, 0.0], [6.335, 0.0], [10, 4.665], [10, 10.0]],
    npoints=201,
    with_manhattan_facing_angles=True,
    with_bbox=True,
    cross_section=cross_section,
) # b_function is a PCell that returns bend components

c = gf.components.cutback_bend90(
    bend90=b_function,
    straight_length=10,
    rows=8,
    columns=3,
    spacing=20,
    cross_section=cross_section,
)
c.show()

```

let me know what you think
Joaquin

**Reply by Rammohan97-boyapati** [reply]
Thank you so much. I love gdsfactory and also thanks for this software and for making it free for all.
